### ISSUE TITLE
Amazon version incompatible with API level 32

### ISSUE THREAD

**Issue by ac941** [RevenueCat/purchases-flutter#389]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-flutter/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-flutter/issues)

It works fine now when compiling my Flutter app and running it on an API Level 30 device or simulator. However the app fails to start on the API Level 32 simulator:

`Failure [INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl27952067.tmp/base.apk (at Binary XML file line #246): com.amazon.device.iap.ResponseReceiver: Targeting S+ (version 31 and above) requires that an explicit value for android:exported be defined when intent filters are present]
`

The reason is that the "android:exported" attribute is missing in the AndroidManifest.xml file. It seems that you are using an old version of "purchases-android", which does not contain that attribute yet. The attribute was added on December 15 2021 and was corrected on December 22 2021 in "purchases-android":

https://github.com/RevenueCat/purchases-android/commits/main/feature/amazon/src/main/AndroidManifest.xml

However the "amazon-latest" version of "purchases-flutter" depends on "purchases-hybrid-common" version "2.0.0-amazon.alpha.4" ( https://github.com/RevenueCat/purchases-flutter/blob/amazon-latest/android/build.gradle ) and "purchases-hybrid-common" version "2.0.0-amazon.alpha.4" depends on "purchases-android" version "5.0.0-amazon.alpha.3" ( https://github.com/RevenueCat/purchases-hybrid-common/blob/2.0.0-amazon.alpha.4/android/build.gradle ). And "purchases-android" version "5.0.0-amazon.alpha.3" does not contain the "export" tags in its AndroidManifest.xml file yet:

https://github.com/RevenueCat/purchases-android/blob/5.0.0-amazon.alpha.3/feature/amazon/src/main/AndroidManifest.xml

Many Flutter libraries nowadays require targetSdkVersion 32. Can you please make purchases-flutter compatible, too?

**Comment by aboedo** [comment]
hi @ac941, thanks for reporting! 

We're currently actively working on updating the Flutter SDK to use the latest purchases-android version under the hood. You can [follow our progress here](https://github.com/RevenueCat/purchases-flutter/pull/354) if you're curious. 

We're in the last phases of testing, if you'd be willing to beta test this version let us know!

**Comment by ac941** [comment]
Great! :-) When do you think it will be finished? And if I would like to beta test, would I just use the "main" branch?

Thank you very much!

**Comment by aboedo** [comment]
Beta testing from the `release/4.0.0` branch should work! 
We're currently fixing a bug where [the purchases overlay might not show up for apps published in Live App Testing](https://github.com/RevenueCat/purchases-android/pulls?q=is%3Apr+is%3Aclosed), we're getting close to shipping once that bug is propagated to flutter. 

The one difference between `release/4.0.0` and `[4.0.0-amazon.alpha.3](https://github.com/RevenueCat/purchases-flutter/releases/tag/4.0.0-amazon.alpha.3)`, is that you no longer need to add the `getAmazonLibrary` task to gradle. If you have already, you should delete it. If you never added it, then you're good 👍 

Note that until we propagate the bug fix I mentioned earlier, Amazon support will only work for the [App Tester](https://developer.amazon.com/docs/in-app-purchasing/iap-install-and-configure-app-tester.html) environment and not Live App Testing. I'll post again here when Live App Testing support is fixed. 

If you give it a shot let me know how it goes!

**Comment by armandojimenez** [comment]
Hi, tried to add the amazon.alpha.3 version to test my app for amazon and I am getting the same error when targeting API 31. As I understand, if I use release/4.0.0 then I can't test the purchase process in live app testing? How can I test it for real? Or I can publish to amazon yet until the bug is fixed?

**Comment by aboedo** [comment]
We're working on delivering another pre-release that fixes Live App Testing for Amazon soon.

In the meantime, `4.0.0-beta.1` is the latest pre-release available. I would recommend holding on publishing until the next beta is available, to avoid issues with Live App Testing and production.

**Comment by armandojimenez** [comment]
I published to live app testing and I am getting this error:
```
PlatformException: PlatformException(2, There was a problem with the store., {code: 2, message: There was a problem with the store., readableErrorCode: StoreProblemError, readable_error_code: StoreProblemError, underlyingErrorMessage: Failed to get user data. Make sure sandbox mode is enabled if using App Tester. https://rev.cat/enter-amazon-sandbox}
```
Is this the error that should show and that you guys are fixing on next beta release right? Asking because right now I have no way to validate if I did the products and code configuration right in my app, since I can't fetch or buy products.

Also, soon as in 1 week, 1 month?

**Comment by armandojimenez** [comment]
> The issues we experienced in Live App Testing were more related to the actual purchase prompt not showing up, I'm not sure this would be the same thing. Did you manage to get [sandbox mode working](https://rev.cat/enter-amazon-sandbox) for the device?

The docs mentions that I need a Fire device and I don't have one currently (Ordered a cheap $30 one with Prime deal, arriving soon) to test that, so I created an APK and uploaded it to live app testing, hoping to test in a Pixel 5 with the Amazon Appstore installed, but I get that error. Is there any way to test without a Fire device that I configured the offerings correctly and that the error I am getting in live app testing is the one that you guys are working on fixing?

**Comment by vegaro** [comment]
I believe you should be able to use the Pixel 5 with the Amazon Appstore installed to test purchases while you wait for your Fire device. Where you using a [Live App Testing](https://developer.amazon.com/docs/in-app-purchasing/iap-testing-overview.html#live-app-testing-lat-mode) build when you got that error? If you are using [App Tester (Sandbox mode) ](https://developer.amazon.com/docs/in-app-purchasing/iap-testing-overview.html#sandbox-mode) you need to enable sandbox mode by running `adb shell setprop debug.amazon.sandboxmode debug`

**Comment by armandojimenez** [comment]
I got it working, the error was that I needed to add the .pem to an asset folder on the android side. I found this in the android setup for amazon in the RC docs, but no mention of it in the Flutter instructions

**Comment by armandojimenez** [comment]
Any timeline on when this is going to hit release? I have my app ready to publush, only waiting on this

**Comment by vegaro** [comment]
We just released an release candidate for 4.0.0 in case you want to give it a try

**Comment by ArmandoJimenezOS** [comment]
I tried that version and published my app, it's currently live.
But I received a crash in sentry, is this because the products were not fetched at the time of purchase?
```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean [android.os.Handler.post](http://android.os.handler.post/)(java.lang.Runnable)' on a null object reference
    at com.revenuecat.purchases.Purchases.dispatch(Purchases.kt:2)
    at com.revenuecat.purchases.Purchases.handleErrorFetchingOfferings(Purchases.kt:9)
    at com.revenuecat.purchases.Purchases.access$handleErrorFetchingOfferings(Purchases.kt:1)
    at com.revenuecat.purchases.Purchases$fetchAndCacheOfferings$2.invoke(Purchases.kt:2)
    at com.revenuecat.purchases.Purchases$fetchAndCacheOfferings$2.invoke(Purchases.kt:1)
    at com.revenuecat.purchases.common.Backend$getOfferings$call$1.onError(Backend.kt:5)
    at com.revenuecat.purchases.common.Dispatcher$AsyncCall.run(Dispatcher.kt:3)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:764)
```

**Comment by vegaro** [comment]
I am going to be closing this issue, since it is fixed in the latest rc version. @armandojimenez's latest comment has its own issue in https://github.com/RevenueCat/purchases-flutter/issues/408
